### PR TITLE
make api accessible via editor, hide tiptap initialization options

### PR DESCRIPTION
**Pull request by YousefED**
This PR:

- removes the difference between BlockNoteEditor and EditorAPI. API methods are directly accessible on blocknoteeditor.
- renames tiptapEditor to _tiptapEditor to clarify it's only for internal use
- We only allow our own `BlockNoteEditorOptions` instead of having it extend from Tiptap's options. This way we can design + document our own API. (and if we want to at some point, we can remove TipTap later)

### CODE REVIEW

**Review by matthewlipski — APPROVED** [review]
Tests are fixed, ready to merge:)

File: packages/core/src/api/nodeConversions/nodeConversions.ts [hunk 1]
```diff
@@ -60,14 +70,19 @@ export function getNodeById(
     if (node.type.name !== "blockContainer" || node.attrs.id !== id) {
       return true;
     }
+    console.log("grdgrdgrdgdr")
```
Comment (YousefED) [inline packages/core/src/api/nodeConversions/nodeConversions.ts:73]: always double check the PR ;)

File: packages/core/test/blockManipulation.test.ts [hunk 2]
```diff
@@ -1,81 +1,111 @@
...
-    const existingBlock = editorAPI.allBlocks[0];
-
-    editorAPI.insertBlocks(multipleBlocks, existingBlock, placement);
-
-    return editorAPI.allBlocks;
-  }
-
   it("Insert before existing block", async () => {
-    const output = insert("before");
+    await waitForEditor()
```
Comment (YousefED) [inline packages/core/test/blockManipulation.test.ts:91]: why is this needed?

Comment (matthewlipski) [reply packages/core/test/blockManipulation.test.ts:91]: When you create a new `BlockNoteEditor()`, that only starts the creation of the editor. It's not ready to be used straight away. For example, if you pass an `onCreate` function where you log smth then immediately after calling `BlockNoteEditor()`, you terminate the app, nothing will get logged.

This is basically what's happening in the tests as well - the editor isn't 100% ready when we start accessing and changing blocks. So this causes, e.g. the UniqueID plugin to not be initialized yet and we see blocks with null IDs.

`waitForEditor()` waits until the editor is actually ready, then the tests continue as normal.
